src/Microsoft.Health.Fhir.Azure/IntegrationDataStore/AzureBlobIntegrationDataStoreClient.cs

```diff
@@ -0,0 +1,256 @@
+
+        public async Task CommitAsync(Uri resourceUri, string[] blockIds, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(resourceUri, nameof(resourceUri));
+            EnsureArg.IsNotNull(blockIds, nameof(blockIds));
+
+            try
+            {
+                await Policy.Handle<StorageException>()
+                    .WaitAndRetryAsync(
+                        retryCount: 3,
+                        sleepDurationProvider: (retryCount) => TimeSpan.FromSeconds(5 * (retryCount - 1)))
+                    .ExecuteAsync(async () =>
+                    {
+                        CloudBlockBlob blob = await GetCloudBlobClientAsync(resourceUri, cancellationToken);
+                        await CommitInternalAsync(blob, blockIds, cancellationToken);
+                    });
+            }
+            catch (StorageException storageEx)
+            {
+                _logger.LogInformation(storageEx, "Failed to commit for {0}", resourceUri);
+
+                HttpStatusCode statusCode = StorageExceptionParser.ParseStorageException(storageEx);
+                throw new IntegrationDataStoreException(storageEx.Message, statusCode);
+            }
+        }
+
+        public async Task AppendCommitAsync(Uri resourceUri, string[] blockIds, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(resourceUri, nameof(resourceUri));
+            EnsureArg.IsNotNull(blockIds, nameof(blockIds));
+
+            try
+            {
+                await Policy.Handle<StorageException>()
+                    .WaitAndRetryAsync(
+                        retryCount: 2,
+                        sleepDurationProvider: (retryCount) => TimeSpan.FromSeconds(5 * (retryCount - 1)))
+                    .ExecuteAsync(async () =>
+                    {
+                        CloudBlockBlob blob = await GetCloudBlobClientAsync(resourceUri, cancellationToken);
+                        await AppendCommitInternalAsync(blob, blockIds, cancellationToken);
+                    });
+            }
+            catch (StorageException storageEx)
+            {
+                _logger.LogInformation(storageEx, "Failed to append commit for {0}", resourceUri);
+
+                HttpStatusCode statusCode = StorageExceptionParser.ParseStorageException(storageEx);
+                throw new IntegrationDataStoreException(storageEx.Message, statusCode);
+            }
+        }
+
+        public async Task<Dictionary<string, object>> GetPropertiesAsync(Uri resourceUri, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(resourceUri, nameof(resourceUri));
+
+            try
+            {
+                return await Policy.Handle<StorageException>()
+                    .WaitAndRetryAsync(
+                        retryCount: 3,
+                        sleepDurationProvider: (retryCount) => TimeSpan.FromSeconds(5 * (retryCount - 1)))
+                    .ExecuteAsync(async () =>
```

Comment:
This looks repeated, perhaps move the policy setup to a function. (e.g. [https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs))

Reply:
Thanks, updated.